VM/vm_params2vm.py

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env python3
+"""
+REQUIREMENTS:
+PATH contains NZVM binary (github:ucgmsim/Velocity-Model/NZVM)
+
+Most basic example:
+
+mpirun -n 3 vm_params2vm.py Hossack ./vm_params.yml ~/Data/VMs
+(The root directory of VMs should be given. ~/Data/VMs/Hossack will be auto-created)
+
+HELP:
+./vm2_params2vm.py -h
+"""
+
+from argparse import ArgumentParser
+from distutils.spawn import find_executable
+
+from logging import Logger
+from multiprocessing import Pool
+import os
+import platform
+
+from shutil import rmtree, copyfile, move
+import subprocess
+from subprocess import Popen
+import sys
+
+from tempfile import mkdtemp
+import yaml
+
+from qcore import constants, qclogging
+from qcore.validate_vm import validate_vm
+
+from common import store_summary, temp_paths
+from gen_coords import gen_coords
+
+NZVM_BIN = find_executable("NZVM")
+
+
+def gen_vm(
+    out_dir,
+    ptemp,
+    vm_threads=1,
+    logger: Logger = qclogging.get_basic_logger(),
+):
+    """
+    Generates VM and relocate the output files from ptemp to out_dir
+    Also generates coordinates files and validates.
+
+    Parameters
+    ----------
+    out_dir : Directory where output files are eventually saved (eg. ..../Data/VMs/Hossack)
+    ptemp : Temporary work directory (eg. .../Data/VMs/Hossack/_tmp_Hossack_nho8ui41)
+    vm_threads : Number of threads to run NZVM binary with if OpenMP-enabled
+    logger :
+    """
+    os.makedirs(out_dir, exist_ok=True)
+    vm_working_dir, nzvm_cfg, vm_params_path = temp_paths(ptemp)
+
+    # NZVM won't find resources if WD is not NZVM dir, stdout not MPROC friendly
+    with open(os.path.join(ptemp, "NZVM.out"), "w") as logfile:
+        logger.debug("Running NZVM binary")
+        nzvm_env = os.environ.copy()
+        nzvm_env["OMP_NUM_THREADS"] = str(vm_threads)
+        nzvm_exe = Popen(
+            [NZVM_BIN, nzvm_cfg],
+            cwd=os.path.dirname(NZVM_BIN),
+            stdout=logfile,
+            env=nzvm_env,
+        )
+        nzvm_exe.communicate()
+    logger.debug("Moving VM files to vm directory")
+    # fix up directory contents
+    logger.debug("Removing Log and Velocity_Model directories")
+    logger.debug("Moving nzvm config and vm_params yaml to vm directory")
+    for f in [
+        os.path.join(vm_working_dir, "Velocity_Model", vm3dfile)
+        for vm3dfile in ["rho3dfile.d", "vp3dfile.p", "vs3dfile.s"]
+    ]:
+        move(f, os.path.join(out_dir, os.path.basename(f)))  # may overwrite
```

Comment:
This seems a bit more complicated/clever than necessary, a single for loop with all the files to be moved in one list would be cleaner